homeassistant/exceptions.py

```diff
@@ -13,6 +13,23 @@ class HomeAssistantError(Exception):
     """General Home Assistant exception occurred."""
 
 
+class ServiceValidationError(HomeAssistantError):
+    """A validation exception occurred when calling a service."""
+
+    def __init__(
+        self,
+        *args: object,
+        domain: str | None = None,
+        translation_key: str | None = None,
+        translation_placeholders: dict[str, str] | None = None,
```

Comment:
Shouldn't we add this directly to `HomeAssistantError`?

Reply:
`HomeAssistantError` is may be too broard. But the way it is implemented now could also be applied to `HomeAssistantError`. In that case may be add an option to suppress the stack trace?

Reply:
Added translation support for `HomeAssistantError` including `ServiceNotFound` (also a subclass of `HomeAssistantError`)